src/components/workers/PoolDetails.tsx

```diff
@@ -103,15 +102,39 @@ const updateWorkerMutation = graphql`
   }
 `;
 
-function PoolDetails(props: PoolDetailsProps) {
+export default function PoolDetails(props: PoolDetailsProps) {
+  const [pool, refetch] = useRefetchableFragment(
+    graphql`
+      fragment PoolDetails_pool on PersistentWorkerPool @refetchable(queryName: "PoolDetailsRefetchQuery") {
+        id
+        name
+        enabledForPublic
+        viewerPermission
+        workers {
+          name
+          disabled
+          hostname
+          version
+          labels
+          ...WorkerStatusChip_worker
+          info {
+            runningTasks {
+              ...TaskStatusChipExtended_task
+            }
+          }
+        }
+      }
+    `,
+    props.pool,
+  );
+
   let [openEditDialog, setOpenEditDialog] = useState(false);
   let [registrationToken, setRegistrationToken] = useState(null);
-  let { pool } = props;
   let classes = useStyles();
 
-  function refetchData() {
-    props.relay.refetch({ poolId: props.pool.id }, { force: true });
-  }
+  const refetchData = useCallback(() => {
```

Comment:
Let's move the "refetch every 10 seconds" logic to `scenes/Workers/PoolById` component and let's use something like [this](https://relay.dev/docs/guided-tour/refetching/refreshing-queries/#when-using-usequeryloader--loadquery) instead of `useRefetchableFragment`.